pkg/products/monitoringspec/reconciler.go

```diff
@@ -276,14 +276,56 @@ func (r *Reconciler) reconcileRoleBindingsForServiceMonitor(ctx context.Context,
 	}
 	//Create role binding for each of the namespace label selectors
 	for _, namespace := range sermon.Spec.NamespaceSelector.MatchNames {
-		err := r.reconcileRoleBinding(ctx, serverClient, namespace)
+		err := r.reconcileRole(ctx, serverClient, namespace)
+		if err != nil {
+			return err
+		}
+		err = r.reconcileRoleBinding(ctx, serverClient, namespace)
 		if err != nil {
 			return err
 		}
 	}
 	return err
 }
 
+func (r *Reconciler) reconcileRole(ctx context.Context,
+	serverClient k8sclient.Client, namespace string) (err error) {
+
+	role := &rbac.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleBindingName,
+			Namespace: namespace,
+		},
+	}
+	opRes, err := controllerutil.CreateOrUpdate(ctx, serverClient, role, func() error {
+		resources := []string{
+			"services",
+			"endpoints",
+			"pods",
+		}
+
+		verbs := []string{
+			"get",
+			"list",
+			"watch",
+		}
```

Comment:
Is the rhmi-operator SA able to grant these permissions in all namespaces? Even if it happens to be able to do it by default, I think maybe we should add those permissions to the ClusterRole specified in deploy/role.yaml. WDYT?

Reply:
Yes, it can grant permissions in all namespaces. Discussed today morning with @matskiv and verified that role.yaml doesnt needed to be updated at this time.

Reply:
not in "all namespaces" actually,
only in the namespaces created by rhmi-operator SA.

It is fairly safe set of permissions IMHO.
But we were trying to keep cluster wide permissions minimal, so I would not add additional cluster permissions if they are not required at the moment.